File: networkutils/src/main/java/net/maxsmr/networkutils/monitor/model/TrafficDirection.java
package net.maxsmr.networkutils.monitor.model;


public enum TrafficDirection {

    NONE(-1), RECEIVE(0), TRANSMIT(1), BOTH(2);

    private final int value;

    TrafficDirection(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }

    public static TrafficDirection fromNativeValue(int value) throws IllegalArgumentException {

        switch (value) {
            case -1:
                return NONE;
            case 0:
                return RECEIVE;
            case 1:
                return TRANSMIT;
            case 2:
                return BOTH;
            default:
                throw new IllegalArgumentException("Incorrect native value for enum type " + TrafficDirection.class.getName() + ": " + value);
        }
    }

}

File: networkutils/src/main/kotlin/net/maxsmr/networkutils/monitor/model/TrafficDirection.kt
package net.maxsmr.networkutils.monitor.model

enum class TrafficDirection(val value: Int) {
    NONE(-1), RECEIVE(0), TRANSMIT(1), BOTH(2);

    companion object {

        fun fromValue(value: Int): TrafficDirection? = try {
            fromValueOrThrow(value)
        } catch (e: IllegalArgumentException) {
            null
        }

        @Throws(IllegalArgumentException::class)
        fun fromValueOrThrow(value: Int): TrafficDirection = when (value) {
                -1 -> NONE
                0 -> RECEIVE
                1 -> TRANSMIT
                2 -> BOTH
                else -> throw IllegalArgumentException("No $value for enum type " + TrafficDirection::class.java.name)
            }
    }
}